src/layer/AtmosphereLayer.js

```diff
@@ -0,0 +1,414 @@
+
+            var gl = dc.currentGlContext,
+                program = dc.currentProgram,
+                skyData = this.skyData,
+                vboId;
+
+            if (!skyData.verticesVboCacheKey) {
+                skyData.verticesVboCacheKey = dc.gpuResourceCache.generateCacheKey();
+            }
+
+            vboId = dc.gpuResourceCache.resourceForKey(skyData.verticesVboCacheKey);
+            if (!vboId) {
+                vboId = gl.createBuffer();
+                dc.gpuResourceCache.putResource(skyData.verticesVboCacheKey, vboId,
+                    this.skyPoints.length * 4);
+                skyData.refreshVertexBuffer = true;
+            }
+
+            gl.bindBuffer(gl.ARRAY_BUFFER, vboId);
+            if (skyData.refreshVertexBuffer) {
+                gl.bufferData(gl.ARRAY_BUFFER, new Float32Array(this.skyPoints), gl.STATIC_DRAW);
+                dc.frameStatistics.incrementVboLoadCount(1);
+                skyData.refreshVertexBuffer = false;
+            }
+
+            gl.vertexAttribPointer(program.vertexPointLocation, 3, gl.FLOAT, false, 0, 0);
+            gl.enableVertexAttribArray(program.vertexPointLocation);
+        };
+
+        // Internal. Intentionally not documented.
+        AtmosphereLayer.prototype.applySkyIndices = function (dc) {
+
+            var gl = dc.currentGlContext,
+                skyData = this.skyData,
+                vboId;
+
+            if (!skyData.indicesVboCacheKey) {
+                skyData.indicesVboCacheKey = dc.gpuResourceCache.generateCacheKey();
+            }
+
+            vboId = dc.gpuResourceCache.resourceForKey(skyData.indicesVboCacheKey);
+            if (!vboId) {
+                vboId = gl.createBuffer();
+                dc.gpuResourceCache.putResource(skyData.indicesVboCacheKey, vboId, this._skyTriStrip.length * 2);
+                skyData.refreshIndicesBuffer = true;
+            }
+
+            gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, vboId);
+            if (skyData.refreshIndicesBuffer) {
+                gl.bufferData(gl.ELEMENT_ARRAY_BUFFER,
+                    new Uint16Array(this._skyTriStrip), gl.STATIC_DRAW);
+                dc.frameStatistics.incrementVboLoadCount(1);
+                skyData.refreshIndicesBuffer = false;
+            }
+        };
+
+        // Internal. Intentionally not documented.
+        AtmosphereLayer.prototype.drawSky = function (dc) {
+
+            var gl = dc.currentGlContext,
+                program = dc.findAndBindProgram(SkyProgram);
+
+            program.loadGlobeRadius(gl, dc.globe.equatorialRadius);
+
+            program.loadEyePoint(gl, dc.navigatorState.eyePoint);
+
+            program.loadVertexOrigin(gl, Vec3.ZERO);
+
+            program.loadModelviewProjection(gl, dc.navigatorState.modelviewProjection);
+
+            program.loadLightDirection(gl, this.activeLightDirection);
+
+            program.setScale(gl);
+
+            gl.depthMask(false);
+            gl.frontFace(gl.CW);
+
```

Comment:
The sky buffer objects for vertices and elements are only used for rendering. Given that, it's unnecessary to keep them around here in client memory. Please remove the class properties for these buffer objects, and simply compute and store them as needed directly into the vertex buffer objects. The result should be applySkyVertices and applySkyIndices.